src/hooks/useLoginForm/useLoginForm.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useEffect } from 'react';
+import { validationSchema } from './validations.js'
+
+
+const useLoginForm = (initialValues) => {
+  const [values, setValues] = useState(initialValues);
+  const [errors, setErrors] = useState(() => {
+    const tempErrors = {}
+    Object.keys(values).forEach(key => tempErrors[key] = null)
+    return tempErrors
+  })
+
+
+  // Private
+
+  const runValidations = (validations, value) => {
+    
+    return validations.map(validation => {
+      const isError = validation.validationFunction(value)
+      return isError ? validation.prompt : isError
+    })
+
+  }
+
+  const getErrorMessage = (validationResults) => (validationResults.filter(result => result != false)[0] || null)
```

Comment:
Looks like this can do the same, a bit more clean
```suggestion
  const getErrorMessage = (validationResults) => validationResults.find(result => !result)
```

Reply:
```suggestion
  const getErrorMessage = (validationResults) => (validationResults.filter(result => result)[0] || null)
```